### ISSUE TITLE
This addon has CPU usage during combat. 

### ISSUE THREAD

**Issue by epigramx** [arkayenro/arkinventory#1529]
It should be at least an option to be eliminated; most users interested in performance would use it.

In case you don't believe it: get the "addon usage" addon and enable "CPU usage".

**Comment by arkayenro** [comment]
i have addon usage installed and with only it and arkinventory installed i get 0.03 ms/s in combat - thats almost nothing.

ai isnt doing much of anything in combat with the bags closed apart from a check to see if you are in-combat, in which case it aborts.

what numbers are you getting?

**Comment by arkayenro** [comment]
even with all my normal mods loaded ai does 2ms/s in combat, pitbull is doing 15ms/s

**Comment by epigramx** [comment]
Yeah it's not dramatic but it exists (which is why someone may want to minmax by just turning off the addon entirely in combat). There's no big need for an instant measure though because you can just reset with the "X" in-combat and get the next second or two. 

![image](https://user-images.githubusercontent.com/2960257/141269546-f6579f96-432f-43f2-b9ab-00f3158b4421.png)

**Comment by arkayenro** [comment]
for starters theres no way in hell im using 18% of your cpu in combat, thats ludicrous and it cannot mean that.

im not trying to be rude but im not sure you realise what youre asking for - how am i supposed to check if youre in combat without using some cpu do do that?

the code is visible, feel free to check what events are listened to, and that they all have "if in combat then abort - or set a flag and then abort so i can run whatever it was when you get out of combat" checks in them or not.  if i've missed one let me know.

ie, there is already minimal stuff being done in combat as is - and dont count the cpu needed to build the bag window in combat if you go and open it while youre in combat, that would be a giant ball of hypocrisy.

**Comment by epigramx** [comment]
> no way in hell im using 18% of your cpu in combat, thats ludicrous and it cannot mean that.

You misread that table. 18% is only referring to the TOTAL of that particular table; the addon itself only does 2.26ms/s on that table which is relatively low (which is why I agreed it's not huge and it's mainly for minmaxing); I'm taking care of those things to that extend so my total CPU usage during combat is not huge (mainly because I trim weakauras and Blizzard lately also appears to disable a lot of LUA features during combat).
